lib/ufe/UsdRotatePivotTranslateUndoableCommand.cpp

```diff
@@ -16,42 +16,40 @@
 
 #include "UsdRotatePivotTranslateUndoableCommand.h"
 #include "private/Utils.h"
+#include "AL/usd/utils/MayaTransformAPI.h"
+#include "../base/debugCodes.h"
 
 MAYAUSD_NS_DEF {
 namespace ufe {
 
-UsdRotatePivotTranslateUndoableCommand::UsdRotatePivotTranslateUndoableCommand(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item)
+UsdRotatePivotTranslateUndoableCommand::UsdRotatePivotTranslateUndoableCommand(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item, const UsdTimeCode& timeCode)
```

Comment:
You'll notice a change here, that's repeated a few times. I need to get a time code from *somewhere*, so at the moment, it just pulls the time value from the current time. input to the proxy shape.

I think the code change here is the correct one to make, but that boots the question of where that time value comes from up to the business logic. At the moment that comes from the time line.

Previously it would be based on the push2prim and readanimatedvalues attrs, and the behaviour could change based on a couple of option vars. Fuck knows if I can fully explain the logic, uhm

Business logic point #1, where does the time come from, when is it the default, when can we read/write from the default, and how do we going about fixing it when we realise we've made the wrong choice(s)?